### PR TITLE
signalfxexporter: reduce flake by increasing timeout

### PR DESCRIPTION
**Pull request by rmfitzpatrick**
**Description:**
Fixing a bug - Flaky test fix where current timeout of 5ms wasn't long enough ci runners (bumping to 500ms).

**Link to tracking Issue:**
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/18696

### CODE REVIEW

File: .chloggen/increasedimensionupatetimeout.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: increase test timeout to reduce flake
+
+# One or more tracking issues related to the change
+issues: [18696]
+
+# (Optional) One or more lines of additional information to render under the primary note.
+# These lines will be padded with 2 spaces and then inserted directly into the document.
+# Use pipe (|) for multiline entries.
+subtext:
```
Comment (dmitryax) [inline .chloggen/increasedimensionupatetimeout.yaml:16]: Don't need a changelog entry for a test fix

File: exporter/signalfxexporter/exporter_test.go [hunk 2]
```diff
@@ -1074,8 +1074,8 @@ func TestConsumeMetadata(t *testing.T) {
 
 			select {
 			case <-c:
-			// wait 5ms longer than send delay
-			case <-time.After(time.Duration(tt.sendDelay)*time.Second + 5*time.Millisecond):
+			// wait 500ms longer than send delay
```
Comment (dmitryax) [inline exporter/signalfxexporter/exporter_test.go:1077]: Maybe `100ms` is enough? It's +0.5 second to every test run

Comment (rmfitzpatrick) [reply exporter/signalfxexporter/exporter_test.go:1077]: I considered 50ms but if it happens again we're pushing more time obligations w/ reruns to contrib devs and would have to bump it again (unless a better fix is apparent). To be fair I think each test used to be 1s until the previous updates. Ok w/ 100ms if you think it's still worth it.

Comment (rmfitzpatrick) [reply exporter/signalfxexporter/exporter_test.go:1077]: To clarify only one test uses the timeout unless bugs have creeped in:

```
=== RUN   TestConsumeMetadataNotStarted
--- PASS: TestConsumeMetadataNotStarted (0.00s)
=== RUN   TestConsumeMetadata
=== RUN   TestConsumeMetadata/Test_property_updates
=== RUN   TestConsumeMetadata/Test_tag_updates
=== RUN   TestConsumeMetadata/Test_quick_successive_updates
=== RUN   TestConsumeMetadata/Test_updates_on_dimensions_with_nonalphanumeric_characters_(other_than_the_default_allow_list)
=== RUN   TestConsumeMetadata/no_dimension_update_for_empty_properties
--- PASS: TestConsumeMetadata (1.51s)
    --- PASS: TestConsumeMetadata/Test_property_updates (0.00s)
    --- PASS: TestConsumeMetadata/Test_tag_updates (0.00s)
    --- PASS: TestConsumeMetadata/Test_quick_successive_updates (1.00s)
    --- PASS: TestConsumeMetadata/Test_updates_on_dimensions_with_nonalphanumeric_characters_(other_than_the_default_allow_list) (0.00s)
    --- PASS: TestConsumeMetadata/no_dimension_update_for_empty_properties (0.50s)
```

Comment (dmitryax) [reply exporter/signalfxexporter/exporter_test.go:1077]: Ok, sounds good. let's keep 500ms given that you made it faster last time :)

**Review by dmitryax — APPROVED** [review]

**Review by codeboten — APPROVED** [review]

**Pull request merged**